### DISCUSSION TITLE
how to handle incorrect detection of ECG events

### DISCUSSION THREAD

**Discussion by mpopopa** [neuropsychology/NeuroKit#640] [General]
Hello- 

I'm new to Neurokit and just tried a few examples to detect ECG events and calculate heart rate. 

It works quite nicely in some data sets, but I found it detect individual heart beats and R peaks incorrectly in some of my data sets. 

So I wonder if anyone had a similar problem and how we should handle it. There is not much option/argument in ecg_process to adjust parameters in ECG event detection. Any thought or comment appreciated. Thank you.

**Comment by DominiqueMakowski** [comment]
You can try with differnet peak-detection algorihms by running nk.ecg_peaks() with different methods
